src/H5Dio.c

```diff
@@ -286,10 +289,20 @@ H5D__read(size_t count, H5D_dset_io_info_t *dset_info)
             dset_info[i].skip_io = false;
         }
 
-        /* Call storage method's I/O initialization routine */
-        if (dset_info[i].layout_ops.io_init &&
-            (dset_info[i].layout_ops.io_init)(&io_info, &(dset_info[i])) < 0)
-            HGOTO_ERROR(H5E_DATASET, H5E_CANTINIT, FAIL, "can't initialize I/O info");
+        /* Check for shared chunk cache support */
+        if (dset_info[i].dset->shared->layout.sc_ops)
```

Comment:
What are the current cases where a dataset will not support the shared chunk cache? It would be fantastic if we could eliminate large amounts of the code that falls in the other path here.

Reply:
Indeed, the idea is to eventually have every dataset type support the shared chunk cache. Currently none of them do, though structured chunk will support it from the beginning